### PR TITLE
SeedDb/Messages

### PR DESCRIPTION
**Pull request by jacobhausler**
- Adds types and relationships for messages and conversations, and SendMessage and StartConversation mutations
 - SeedDb mutation seeds 5 locations that each have 8 lists with between 4-12 categories

### CODE REVIEW

**Comment by jacobhausler** [comment]
crap wait this also includes all the changes from the messages branch

File: src/modules/conversations/mutations.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    toUserId: types.ID,
+  },
+  outputFields: {
+    changedConversation: types.Conversation,
+  },
+  mutateAndGetPayload: async (
+    { subject, refPostId, toUserId },
+    { db, session: { currentUserId } }
+  ) => {
+    assert(subject, 'Conversations must have subjects');
```
Comment (slightlytyler) [inline src/modules/conversations/mutations.js]: `subject` singular

File: src/modules/conversations/mutations.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    assert(toUserId, 'Conversations must have a toUserId');
+
+    // will fail if toUser doesn't exist
+    const toUser = await db.findBy('users', { id: stripId(toUserId) });
+    // if refPostId has a value, this will fail if the post doesn't exist
+    if (!isNil(refPostId)){
+      const refPost = await db.findBy('posts', { id: stripId(refPostId) });
```
Comment (slightlytyler) [inline src/modules/conversations/mutations.js]: What is this block doing? Don't see `refPost` used anywhere else... curious as the linting passed

File: src/modules/conversations/mutations.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    // create and clean the insertFields
+    const insertFields = chain({
```
Comment (slightlytyler) [inline src/modules/conversations/mutations.js]:
```javascript
const insertFields = omitBy(
  {
    createdAt: new Date(),
    updatedAt: new Date(),
    referencedPostId: stripId(refPostId),
    subject,
  },
 isNil
);
```
No need for chain here as where only performing one iterative operation

File: src/modules/conversations/mutations.js (hunk 2 above)
Comment (jacobhausler) [inline src/modules/conversations/mutations.js]: we might be able to remove the "const refPost ="; the function makes sure that if a postId was supplied, it's a legit post

File: src/modules/conversations/mutations.js [hunk 4]
```diff
@@ -23,16 +23,17 @@ export const Start = types => ({
...
-      updatedAt: new Date(),
-      referencedPostId: stripId(refPostId),
-      subject,
-    }).omitBy(isNil).value();
+    const insertFields = omitBy({
+        createdAt: new Date(),
+        updatedAt: new Date(),
+        referencedPostId: stripId(refPostId),
+        subject,
+      },isNil
```
Comment (slightlytyler) [inline src/modules/conversations/mutations.js]: Miss space before the commma, the linter must be not working...

File: src/modules/conversations/mutations.js [hunk 5]
```diff
@@ -23,16 +23,17 @@ export const Start = types => ({
     const toUser = await db.findBy('users', { id: stripId(toUserId) });
     // if refPostId has a value, this will fail if the post doesn't exist
     if (!isNil(refPostId)){
-      const refPost = await db.findBy('posts', { id: stripId(refPostId) });
+      await db.findBy('posts', { id: stripId(refPostId) });
```
Comment (slightlytyler) [inline src/modules/conversations/mutations.js]: Why do you need need to get the post here?

**Comment by slightlytyler** [comment]
LGTM

**Pull request merged**